src/spaceObjects/spaceshipParts/weaponTube.cpp

```diff
@@ -91,6 +92,7 @@ void WeaponTube::fire(float target_angle)
     if (parent->docking_state != DS_NotDocking) return;
     if (parent->current_warp > 0.0) return;
     if (state != WTS_Loaded) return;
+    has_fired = true;
```

Comment:
This won't work properly on clients (as the fire function is only called on the server)

Reply:
I guess I'm missing something here as I don't see why the client should be allowed to set this value at all.  Perhaps the entire function should be guarded inside a `if (game_server)` check?

Reply:
Obviously I didn't initially comprehend what you meant.... I can be a bit daft sometimes...